### PR TITLE
Fix various logical bugs (found by error-prone)

### PR DESCRIPTION
**Pull request by sixlettervariables**
This fixes a few logical bugs found while playing with [Google's error-prone](https://github.com/google/error-prone).

The only one that I'm not 100% sure of is the last commit, where `tonnage` was not passed to the other EquipmentPart ctor. It looks like it is used as if tonnage means the part's tonnage in some cases, and the unit's tonnage in other cases (or `0`).

### CODE REVIEW

**Comment by HammerGS** [comment]
This pull request **fixes 1 alert** when merging 486473c25fdf14ef858e42d56e255096835bd1b3 into ae86420592ca244d37340bf8859dfd4b33b1f89c - [view on LGTM.com](https://lgtm.com/projects/g/MegaMek/mekhq/rev/pr-cd0797ecdfae27448a9db24ddd9873f7645b5d9b)

**fixed alerts:**

* 1 for Unused format argument

---

*Comment posted by [LGTM.com](https://lgtm.com)*

File: MekHQ/src/mekhq/campaign/Campaign.java [hunk 1]
```diff
@@ -2578,7 +2578,7 @@ public void fixPart(IPartWork partWork, Person tech) {
         if (roll >= target.getValue()) {
             report = report + partWork.succeed();
             if (getCampaignOptions().payForRepairs()
-                    && action == " fix "
+                    && action.equals(" fix ")
```
Comment (VicenteCartas) [inline MekHQ/src/mekhq/campaign/Campaign.java]: Wouldn't we want this as a constant?

Comment (sixlettervariables) [reply MekHQ/src/mekhq/campaign/Campaign.java]: Yes, going for minimal changes in each area since this is an automated analysis fix-up.

**Pull request merged**